CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/utils/MediaStoreUtils.kt

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.example.cameraxbasic.utils
+
+import android.content.ContentUris
+import android.content.Context
+import android.database.Cursor
+import android.net.Uri
+import android.os.Build
+import android.provider.MediaStore
+import androidx.core.net.toUri
+import java.io.File
+
+
+/**
+ * A utility class for accessing this app's photo storage.
+ *
+ * Since this app doesn't request any external storage permissions, it will only be able to access
+ * photos taken with this app. If the app is uninstalled, the photos taken with this app will stay
+ * on the device, but reinstalling the app will not give it access to photos taken with the app's
+ * previous instance. You can request further permissions to change this app's access. See this
+ * guide for more: https://developer.android.com/training/data-storage.
+ */
+class MediaStoreUtils(private val context: Context) {
+
+    // Suppress DATA index deprecation warning since we need the file location for the Glide library
+    @Suppress("DEPRECATION")
+    private val imageDataColumnIndex = MediaStore.Images.Media.DATA
+    private val imageIdColumnIndex = MediaStore.Images.Media._ID
+    val mediaStoreCollection: Uri? = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+        MediaStore.Images.Media.getContentUri(MediaStore.VOLUME_EXTERNAL)
+    } else {
+        context.getExternalFilesDir(null)?.toUri()
+    }
+
+    private fun getMediaStoreImageCursor(mediaStoreCollection: Uri): Cursor? {
```

Comment:
Should these async operations be converted to `suspend fun` and switch to the correct dispatcher?
ie.
```
private suspend fun getMediaStoreImageCursor(mediaStoreCollection: Uri): Cursor? =
    withContext(Dispatchers.IO) {
        ...
    }
```
That way it's clear to the callsite that this is an async operation and it internally ensures that the operation is performed on the correct thread
And the result would be returned on the calling thread which could be UI thread which should eliminate doing something like `someUiElement.post { ... }`

Reply:
I believe I addressed this concern, but please take another look and let me know.